Galaxus.Functional.Tests/Option/Async/AsyncOptionExtensions.MatchAsyncTest.cs

```diff
@@ -0,0 +1,446 @@
+﻿using System;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+namespace Galaxus.Functional.Tests.Option.Async;
+
+[TestFixture]
+public class MatchAsyncTest
+{
+    [Test]
+    public async Task Option_MatchAsyncSome_Works()
+    {
+        var some = "hello".ToOption();
+
+        {
+            var called = false;
+            await some.MatchAsync(async _ =>
+            {
+                called = true;
+                await Task.CompletedTask;
+            }, async () =>
+            {
+                await Task.CompletedTask;
+            });
+            Assert.IsTrue(called);
+        }
+
+        {
+            var called = false;
+            await some.MatchAsync(_ =>
+            {
+                called = true;
+            }, async () => { await Task.CompletedTask; });
+            Assert.IsTrue(called);
+        }
+
+        {
+            var called = false;
+            await some.MatchAsync(async _ =>
+                {
+                    called = true;
+                    await Task.CompletedTask;
+                },
+                () => { });
+            Assert.IsTrue(called);
+        }
+
+        {
+            var called = false;
+            var number = await some.MatchAsync(async _ =>
+                {
+                    called = true;
```

Comment:
It's a small nit, but we don't really need the `called`-bit when we work with the value-returning match; we can just check the value and know which one was called.